src/posthog-surveys.ts

```diff
@@ -155,4 +161,101 @@ export class PostHogSurveys {
+                // :KLUDGE: for now, look up the choiceIndex based on the response
+                // TODO: once QuestionTypes.MultipleChoiceQuestion is refactored, pass the selected choiceIndex into this method
+                const selectedChoiceIndex = question.choices.indexOf(`${response}`)
+
+                if (question.branching?.responseValues?.hasOwnProperty(selectedChoiceIndex)) {
+                    const nextStep = question.branching.responseValues[selectedChoiceIndex]
+
+                    // Specific question
+                    if (Number.isInteger(nextStep)) {
+                        return nextStep
+                    }
+
+                    if (nextStep === SurveyQuestionBranchingType.ConfirmationMessage) {
+                        return SurveyQuestionBranchingType.ConfirmationMessage
+                    }
+
+                    return nextQuestionIndex
+                }
+            } else if (question.type === SurveyQuestionType.Rating) {
+                let responseValueToBucket: { [key: number]: string } = {}
+
+                if (question.scale === 3) {
+                    responseValueToBucket = {
+                        1: 'negative',
+                        2: 'neutral',
+                        3: 'positive',
+                    }
+                } else if (question.scale === 5) {
+                    responseValueToBucket = {
+                        1: 'negative',
+                        2: 'negative',
+                        3: 'neutral',
+                        4: 'positive',
+                        5: 'positive',
+                    }
+                } else if (question.scale === 10) {
+                    responseValueToBucket = {
+                        0: 'detractors',
+                        1: 'detractors',
+                        2: 'detractors',
+                        3: 'detractors',
+                        4: 'detractors',
+                        5: 'detractors',
+                        6: 'detractors',
+                        7: 'passives',
+                        8: 'passives',
+                        9: 'promoters',
+                        10: 'promoters',
+                    }
+                }
+
+                const resNumber = Number(response)
+                const ratingBucket = responseValueToBucket[resNumber]
+
+                if (question.branching?.responseValues?.hasOwnProperty(ratingBucket)) {
+                    const nextStep = question.branching.responseValues[ratingBucket]
+
+                    // Specific question
+                    if (Number.isInteger(nextStep)) {
+                        return nextStep
+                    }
+
+                    if (nextStep === SurveyQuestionBranchingType.ConfirmationMessage) {
+                        return SurveyQuestionBranchingType.ConfirmationMessage
+                    }
+
+                    return nextQuestionIndex
+                }
+            }
+
+            return nextQuestionIndex
```

Comment:
I think `console.warn`s whenever we fallback because of invalid input would be useful - there's no case I can see where if things are setup correctly we reach here

Reply:
Looks good! Are there any other cases where it's an unexpected return, or are they all for the implicit next question case?